### PR TITLE
Middleware is uncountable

### PR DESCRIPTION
**Pull request by l-vo**
|      Q       |   A
|------------- | -----------
| Type         | typo
| Fixed issues | 

#### Summary

According to https://www.oxfordlearnersdictionaries.com/definition/english/middleware, middleware is uncountable.

### CODE REVIEW

File: src/Configuration.php [hunk 1]
```diff
@@ -212,21 +212,47 @@ public function getAutoCommit(): bool
...
-     * @param Middleware[] $middlewares
+     * @deprecated setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.
+     *
+     * @param Middleware[] $middleware
+     *
+     * @return $this
+     */
+    public function setMiddlewares(array $middleware): self
+    {
+        trigger_error('setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.', E_USER_DEPRECATED);
```
Comment (derrabus) [inline src/Configuration.php]: DBAL 3.6 has been released already. We cannot introduce new deprecation in that version. All changes that deprecate stuff have to be moved to 3.7.x!

Comment (derrabus) [inline src/Configuration.php]: We use the doctrine/deprecations library in this repository instead of calling `trigger_error()` directly.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Configuration.php (hunk 1 above)
Comment (l-vo) [reply src/Configuration.php]: Oh yes my bad. Thank you :)

Comment (l-vo) [reply src/Configuration.php]: Indeed, I didn't pay attention to that. Thank you.

**Comment by l-vo** [comment]
@derrabus updated.

File: docs/en/reference/architecture.rst [hunk 2]
```diff
@@ -43,8 +43,8 @@ interfaces:
 -  ``Doctrine\DBAL\Driver\Statement``
 -  ``Doctrine\DBAL\Driver\Result``
 
-Middlewares
------------
+Middleware
+----------
 
 A middleware sits in the middle between the wrapper components and the driver.
```
Comment (derrabus) [inline docs/en/reference/architecture.rst:49]: If middleware is not countable, is there something like _a_ middleware? The way you use the term in the `Configuration` class now, middleware is the whole layer between the driver an the wrapper. I think, we need to find a different term for the thing that we create through implementing the `Middleware` interface.

**Review by derrabus — CHANGES_REQUESTED** [review]
Deprecations have to be documented in the `UPGRADE.md` file.

**Comment by derrabus** [comment]
Regarding the deprecation, I'm really wondering if it's worth it. While the current method naming pattern `[gs]etMiddlewares()` is grammatically incorrect as I've learned, changing it at this point only solves an academic/esthetic issue and generates avoidable busy work downstream.

Maybe it's enough to adjust the documentation only and keep the method names as they are?

Paging @morozov as he's the main author of the middleware system.

**Comment by l-vo** [comment]
@derrabus I'm agree, but even if it's not a good Idea, this PR will keep a trace of this discussion 👍

**Comment by morozov** [comment]
I don't think this change provides any value while at the same it clearly causes damage. The definition from the dictionary focuses on the middleware being a layer (a collective notion) while in this specific design, middleware is an interface and its specific implementations, so they are countable.

The existing code operates both the specific instances of middleware and their collections, so using the same single tense for both might become confusing and impractical at some point.

Natural languages evolve and adjust to their actual usage, they are not set in stone. Here are the examples of using "[middlewares](https://sourcegraph.com/search?q=middlewares&groupBy=repo)" in the open-source code.

**Comment by l-vo** [comment]
@morozov it makes sense, thank you for your feedback 👌

**Pull request closed without merging**